src/main/java/de/sldk/mc/metrics/Entities.java

```diff
@@ -2,22 +2,50 @@
 
 import io.prometheus.client.Gauge;
 import org.bukkit.World;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
 import org.bukkit.plugin.Plugin;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public class Entities extends WorldMetric {
 
     private static final Gauge ENTITIES = Gauge.build()
             .name(prefix("entities_total"))
             .help("Entities loaded per world")
-            .labelNames("world")
+            .labelNames("world", "type", "alive")
             .create();
 
+    /**
+     * Override the value returned by {@link EntityType#isAlive()}.
+     */
+    private static final Map<EntityType, Boolean> aliveOverride = new HashMap<EntityType, Boolean>() {{
+        put(EntityType.ARMOR_STAND, false);
+    }};
+
+
     public Entities(Plugin plugin) {
         super(plugin, ENTITIES);
     }
 
     @Override
     public void collect(World world) {
-        ENTITIES.labels(world.getName()).set(world.getEntities().size());
+        Map<EntityType, Long> mapEntityTypesToCounts = world.getEntities().stream()
+                .collect(Collectors.groupingBy(Entity::getType, Collectors.counting()));
+
+        mapEntityTypesToCounts.keySet()
+                .forEach(entityType ->
+                        ENTITIES
+                                .labels(world.getName(),
+                                        entityType.name(),
```

Comment:
I'm wondering if we should use `entityType.getKey().getKey()` here instead of the enum name. I think those strings are closer to the real names in mc but eh - not really important :man_shrugging: 

Do you have an opinion on this?

Reply:
Good question! It would be nicer to have things match what's shown in game, and lowercase is nicer than ALL CAPS.

Downsides are 
* What happens if the keys clash? Could two mods add two things with the same name?
* special handling is required for types without keys, e.g. `EntityType.UNKNOWN` (otherwise there's an error: `IllegalArgumentException: EntityType doesn't have key! Is it UNKNOWN?`).

For more info, here are the entities where (ignoring case) name and key don't match:

* DROPPED_ITEM - item
* ENDER_CRYSTAL - end_crystal
* ENDER_SIGNAL - eye_of_ender
* FIREWORK - firework_rocket
* FISHING_HOOK - fishing_bobber
* LEASH_HITCH - leash_knot
* LIGHTNING - lightning_bolt
* MINECART_CHEST - chest_minecart
* MINECART_COMMAND - command_block_minecart
* MINECART_FURNACE - furnace_minecart
* MINECART_HOPPER - hopper_minecart
* MINECART_MOB_SPAWNER - spawner_minecart
* MINECART_TNT - tnt_minecart
* MUSHROOM_COW - mooshroom
* PIG_ZOMBIE - zombie_pigman
* PRIMED_TNT - tnt
* SNOWMAN - snow_golem
* SPLASH_POTION - potion
* THROWN_EXP_BOTTLE - experience_bottle

I'll think about it...

Reply:
I've made the change to use `getKey()` for now but I'll see how that works out.

Reply:
Hm, I think it's fine the way it is now :+1: